patches/0117-libavfilter-dnn-dnn_backend_torch-Add-async-to-torch.patch

```diff
@@ -0,0 +1,127 @@
+From 6637e8e45f73f7263a20b724fe8369b9fa1d8660 Mon Sep 17 00:00:00 2001
+From: Wenbin Chen <wenbin.chen@intel.com>
+Date: Tue, 14 Nov 2023 16:38:52 +0800
+Subject: [PATCH 1/2] libavfilter/dnn/dnn_backend_torch: Add async to torch
+ backend
+
+Signed-off-by: Wenbin Chen <wenbin.chen@intel.com>
+---
+ libavfilter/dnn/dnn_backend_torch.cpp | 54 +++++++++++++++++++--------
+ 1 file changed, 38 insertions(+), 16 deletions(-)
+
+diff --git a/libavfilter/dnn/dnn_backend_torch.cpp b/libavfilter/dnn/dnn_backend_torch.cpp
+index f54493d742..855b1b3b97 100644
+--- a/libavfilter/dnn/dnn_backend_torch.cpp
++++ b/libavfilter/dnn/dnn_backend_torch.cpp
+@@ -40,6 +40,8 @@ typedef struct THOptions{
+     char *device_name;
+     c10::DeviceType device_type;
+     int optimize;
++    uint8_t async;
++    uint32_t nireq;
+ } THOptions;
+ 
+ typedef struct THContext {
+@@ -76,6 +78,8 @@ typedef struct THRequestItem {
+ static const AVOption dnn_th_options[] = {
+     { "device", "device to run model", OFFSET(options.device_name), AV_OPT_TYPE_STRING, { .str = "cpu" }, 0, 0, FLAGS },
+     { "optimize", "turn on graph executor optimization", OFFSET(options.optimize), AV_OPT_TYPE_INT, { .i64 = 0 }, 0, 1, FLAGS},
++    { "nireq",           "number of request",             OFFSET(options.nireq),           AV_OPT_TYPE_INT,    { .i64 = 0 },     0, INT_MAX, FLAGS },
++    { "async",           "use DNN async inference",       OFFSET(options.async),           AV_OPT_TYPE_BOOL,   { .i64 = 1 },     0,       1, FLAGS },
```

Comment:
How about using single option instead ?
when async (or async_depth) = 0 , use sync mode, when async (or async_depth) > 0, use async mode, the value is the number of request

Reply:
I want to keep same parameter with openvino. There is a macro definition for common options I forget to use it. I change the code to use it.